### ISSUE TITLE
I get the output of filters in layer2 all grey

### ISSUE THREAD

**Issue by tearhupo121031** [utkuozbulak/pytorch-cnn-visualizations#10]
sorry for bother you, I run the code 'cnn_layer_visualization' and observe the output of all the filters in layer2, but i get the picture all grey, and the loss decay very slow:
![2018-05-22 10-21-02](https://user-images.githubusercontent.com/28970557/40338858-220e146a-5daa-11e8-9889-3a70050efecd.png)
![2018-05-22 10-21-36](https://user-images.githubusercontent.com/28970557/40338860-297290fa-5daa-11e8-9bdb-1d93682e290b.png)
is my iteration too small?

**Comment by tearhupo121031** [comment]
I change the epoch about 5000, and get the correct pictures, thank you for help!

**Comment by deepblue0822** [comment]
my results are all the same,what happened?

**Comment by utkuozbulak** [comment]
It is a problem with the hyperparameters, also the way maximization/minimization works. Initially, I made an error with min/max, see #5 for the discussion on that. I made small changes in the code which then broke how the parameters affect the optimization which made visualizing earlier layers really hard.

I just updated the optimizer from SGD to Adam and changed the lr and iterations.
https://github.com/utkuozbulak/pytorch-cnn-visualizations/blob/ff2ec0c028b1c8e8839796895b83016d44f4be5a/src/cnn_layer_visualization.py#L48

with this change, even if you don't change any parameters you will be able to visualize different layers with ease in only around 30 epochs (you can play around lr and reduce this even further). 

Here are some examples without changing any parameter, only the targeted layer:

Layer 24:
![layer_vis_l24_f100_iter30](https://user-images.githubusercontent.com/19213588/41080635-6e7cb7ec-6a62-11e8-8edb-cf9ec6703303.jpg)
![layer_vis_l24_f50_iter30](https://user-images.githubusercontent.com/19213588/41080643-75eaf20a-6a62-11e8-86c4-3956bed879fe.jpg)

Layer 17:
![layer_vis_l17_f25_iter30](https://user-images.githubusercontent.com/19213588/41080652-7cb3fffa-6a62-11e8-9086-883847d717e5.jpg)
![layer_vis_l17_f5_iter30](https://user-images.githubusercontent.com/19213588/41080682-8f2f284e-6a62-11e8-9ee5-464f907d3642.jpg)


Layer 10:
![layer_vis_l10_f9_iter25](https://user-images.githubusercontent.com/19213588/41080691-9915004a-6a62-11e8-9c8e-545449b94243.jpg)

![layer_vis_l10_f10_iter30](https://user-images.githubusercontent.com/19213588/41080685-936a60cc-6a62-11e8-866a-630f183a9938.jpg)

Layer 2:

![layer_vis_l2_f15_iter40](https://user-images.githubusercontent.com/19213588/41080696-a095efb4-6a62-11e8-8e99-55dd0b76a17d.jpg)

![layer_vis_l2_f8_iter25](https://user-images.githubusercontent.com/19213588/41080702-a53695c8-6a62-11e8-86c4-2b33474a5caa.jpg)

I hope this solves your problems.
